Increase MAX_CPUS to 512

lucashc opened the pull request:
Increase MAX_CPUS from 256 to 512 to support higher cpu count systems.
Fixes #485

tohojo reviewed (changes requested):
Thanks for the fix! Please add the information from the issue about which systems this affects to the commit message, and add a signed-off-by line as well.

lucashc commented:
I added the requested changes and updated the commit message.
One more thought: The Linux kernel allows many more CPUs than just 512. For instance, the kernel of my Ubuntu 24.04 LTS has been configured to support 8192 CPUs. Should we increase it even further?
On the other hand, most common servers will not exceed 512 CPUs.

tohojo reviewed (approved):
Looks good, thanks!

I don't think we should increase it further as this adds overhead for all systems (being a compile-time value). If we need to scale to a higher number of CPUs we should start sizing things dynamically at runtime (or maybe find some way to avoid the per-CPU stuff if possible).
